### ISSUE TITLE
pacaur fails to build when makepkg succeeds

### ISSUE THREAD

**Issue by jpkotta** [rmarquis/pacaur#313]
The elixir package in AUR does not build with pacaur, but works fine with makepkg.  Elixir used to build just fine with pacaur; the latest version is the first I've had trouble with.  

----

$ pacaur -S elixir
:: resolving dependencies...
:: looking for inter-conflicts...

AUR Packages  (1): elixir-1.0.3-1  

:: Proceed with download? [Y/n] 

:: Retrieving package(s)...
:: Checking elixir integrity...
==> Making package: elixir 1.0.3-1 (Fri Feb 20 10:40:57 CST 2015)
==> WARNING: Skipping dependency checks.
==> Retrieving sources...
  -> Found v1.0.3
==> Validating source files with sha512sums...
    v1.0.3 ... Passed
==> Extracting sources...
  -> Extracting v1.0.3 with bsdtar
==> Starting prepare()...
==> Sources are ready.
:: Building elixir package(s)...
==> Making package: elixir 1.0.3-1 (Fri Feb 20 10:40:58 CST 2015)
==> Checking runtime dependencies...
==> Checking buildtime dependencies...
==> WARNING: Using existing $srcdir/ tree
==> Removing existing $pkgdir/ directory...
==> Starting build()...
make: *** No targets specified and no makefile found.  Stop.
==> ERROR: A failure occurred in build().
    Aborting...
:: elixir cleaning skipped
:: failed to build elixir package(s)

----

$ pacaur -d elixir
:: elixir downloaded to /tmp/pacaurtmp-jpkotta
$ cd /tmp/pacaurtmp-jpkotta/elixir/
$ makepkg
==> Making package: elixir 1.0.3-1 (Fri Feb 20 10:48:46 CST 2015)
==> Checking runtime dependencies...
==> Checking buildtime dependencies...
==> Retrieving sources...
  -> Downloading v1.0.3...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   133    0   133    0     0    293      0 --:--:-- --:--:-- --:--:--   292
100 1014k  100 1014k    0     0  1029k      0 --:--:-- --:--:-- --:--:-- 1029k
==> Validating source files with sha512sums...
    v1.0.3 ... Passed
==> Extracting sources...
  -> Extracting v1.0.3 with bsdtar
==> Starting prepare()...
==> Starting build()...
==> elixir (compile)
Compiled src/elixir_parser.yrl
Compiled src/elixir_utils.erl
Compiled src/elixir_try.erl
Compiled src/elixir_translator.erl
...

----

$ pacaur -v
pacaur 4.2.19
$ cat ~/.config/pacaur/config 
editpkgbuild=false
editinstall=false
clean=false
sudoloop=True

**Comment by rmarquis** [comment]
I can reproduce. Will have a look shortly.

**Comment by rmarquis** [comment]
This happens because of the --noextract flag that is passed to makepkg (-e). I'm not exactly sure why this gives an error, but I pushed a fix in 0600ee929c68e3dd37ea2cad31dee5f88347b4f1.

You can install pacaur-git and try it.

**Comment by jpkotta** [comment]
That makes sense.  prepare() sets ELIXIR_SRC_DIR, which is then used in build(), but prepare() isn't run with --noextract.  I think the more correct fix is to not assume ELIXIR_SRC_DIR is global in the PKGBUILD, so I'll ping the maintainer.  In any event, pacaur-git works for me now.

**Comment by rmarquis** [comment]
> That makes sense. prepare() sets ELIXIR_SRC_DIR

Right. I overlooked the PKGBUILD and did miss that. The PKGBUILD should be fixed, as the prepare() section shouldn't be used to set temporary variables (that's just ugly).

However, if using `--verify` in the integrity check part of pacaur makes more sense than using the hacky "makepkg -od", this kinda breaks the "--needed" function with VCS packages (see 9bcf7ce9bf31287b55b97bb5732bccd1c71bc4d0 for the reason I've used `-od` in the first place).

I'll try to fix this VCS packages rebuild issue in a better way.
